tools/gh-release.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+"""
+Draft a release in GitHub by calling into its API.
+
+Assumes all the releases are in the current path.
+"""
+
+import argparse
+import json
+import subprocess
+import tarfile
+from pathlib import Path
+
+from github import Github
+
+
+def handler_github_release(event, context):
+    print(json.dumps(event))
+    # Parameters from event / environment
+    tag_version = event["tag_version"]
+    repo = event["repo"]
+    github_token = event["github_token"]
+    prerelease = False
+
+    # Do not include signatures in GitHub release since we aren't
+    # making those keys public
+    exclude_files = {"RELEASE_NOTES", "SHA256SUMS.sig"}
+
+    assets = []
+    for arch in ["x86_64", "aarch64"]:
+        release_dir = Path(f"release-{tag_version}-{arch}")
+        # Build tarball
+        release_tgz = Path(f"firecracker-{tag_version}-{arch}.tgz")
+        print(f"Creating release archive {release_tgz} ...")
+        with tarfile.open(release_tgz, "w:gz") as tar:
+            files = [x for x in release_dir.rglob("*") if x.is_file()]
+            for asset in files:
+                if asset.name in exclude_files:
+                    print(f"Skipping file {asset}")
+                    continue
+                if asset.name.endswith(arch):
+                    print(f"Setting +x bit for {asset}")
+                    asset.chmod(0o755)
+                print(f"Adding {asset} to {release_tgz}")
+                tar.add(asset)
+        print("Done. Archive successfully created. sha256sum result:")
+        sha256sums = release_tgz.with_suffix(release_tgz.suffix + ".sha256.txt")
+        subprocess.run(
+            f"sha256sum {release_tgz} | awk '{{print $1}}' > {sha256sums}",
+            check=True,
+            shell=True,
+        )
+        print(sha256sums.read_text("utf-8"))
+        assets.append(release_tgz)
+        assets.append(sha256sums)
+
+    message_file = Path(f"release-{tag_version}-x86_64") / "RELEASE_NOTES"
```

Comment:
aside: fascinated by use of operator overloading 😂

Reply:
It's really handy :)